pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/MethodTypeResolution.java

```diff
@@ -510,4 +510,12 @@ public static boolean isSubtypeable(JavaTypeDefinition parameter, JavaTypeDefini
 
         return false;
     }
+
+    public static JavaTypeDefinition boxPrimitive(JavaTypeDefinition def) {
+        if (!def.isPrimitive()) {
+            return null;
+        }
+
+        return JavaTypeDefinition.forClass(BOXED_PRIMITIVE_SUBTYPE_ORDER.get(PRIMITIVE_SUBTYPE_ORDER.indexOf(def.getType())));
```

Comment:
LinkedHashMap is the one that maintains insertion order, but it only exposes it by iteration. That would require a loop to find the insertion index of an entry by manual counting. isMethodConvertible uses primitive to boxed and boxed to primitive as well.
We can add two more maps to simplify code, but I see no way around the two lists.